spark/v3.1/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -566,12 +578,14 @@ private static class WriterFactory implements DataWriterFactory, StreamingDataWr
     protected WriterFactory(
         Broadcast<Table> tableBroadcast,
         FileFormat format,
+        PartitionSpec outputSpec,
```

Comment:
How about passing spec id and looking up PartitionSpec on the last part?  It will be less to serailize (I think WriterFactory is serialized and sent to executors)

Reply:
Got it, I'll add that shortly.

Reply:
Done. Changed it on all Spark versions.